components/sapi/CMakeLists.txt

```diff
@@ -2,7 +2,7 @@ add_library(datadog_php_sapi sapi.c)
 
 target_include_directories(datadog_php_sapi
   PUBLIC
-    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../..>
```

Comment:
Was this the secret to exposing the parent directory name all along?

Reply:
Here's what the variables refer to:
```
components    <- PROJECT_SOURCE_DIR
└── sapi      <- CMAKE_CURRENT_SOURCE_DIR
```

We need `components/` to _not_ be in the include path, because then any component can be referred to by `component/component.h` instead of `components/component/component.h`.

We could probably do some build-time copies of the files so that each component only exposes its own headers and such, but that's a quality of implementation detail that could be changed later if we feel the need to safe-guard it.

Reply:
I like it. If we all agree, I think we'll eventually be able to change ZAI headers too. Although, the path would be a little different from [our former attempts/discussions](https://github.com/DataDog/dd-trace-php/blob/5f37d0e/zend_abstract_interface/functions/CMakeLists.txt#L28). I think `zend_abstract_interface/component/component.h` would have little chance of colliding with other libs. :)